Library/Homebrew/os/linux/global.rb

```diff
@@ -1,7 +1,10 @@
 # frozen_string_literal: true
 
 # enables experimental patchelf.rb write support.
-HOMEBREW_PATCHELF_RB_WRITE = ENV["HOMEBREW_PATCHELF_RB_WRITE"].present?.freeze
+HOMEBREW_PATCHELF_RB_WRITE = (
+  ENV["HOMEBREW_PATCHELF_RB_WRITE"].present? || ENV["HOMEBREW_DEVELOPER"].present? &&
+  !(ENV["CI"].present? || ENV["NO_HOMEBREW_PATCHELF_RB_WRITE"].present?)
```

Comment:
```suggestion
  ENV["HOMEBREW_PATCHELF_RB_WRITE"].present?
  || ENV["HOMEBREW_DEVELOPER"].present?
  && !ENV["CI"].present?
  && !ENV["HOMEBREW_NO_PATCHELF_RB_WRITE"].present?)
```

Reply:
i'll separate the &s.
ruby has problems parsing operators on next line, 
even i prefer
A
&& b
|| c

over 
A &&
b ||
c